device-types/Juniper/PTX10001-MR.yml

```diff
@@ -0,0 +1,165 @@
+---
+manufacturer: Juniper
+model: PTX10001-MR
+slug: juniper-ptx10001-mr
+u_height: 1.0
+is_full_depth: true
+airflow: rear-to-front
+comments: '[Juniper PTX10001 Data Sheet](https://www.juniper.net/us/en/products/routers/ptx-series/ptx10001-36mr-packet-transport-router.html)'
+console-ports:
+  - name: Console
+    type: rj-45
+power-ports:
+  - name: PSU0
+    type: saf-d-grid
+  - name: PSU1
+    type: saf-d-grid
```

Comment:
It appears that these PSUs are actually modular. Same with the fans

Thus, you should be defining `module-bays` for the PSUs (and FAN slots). Then, you should ensure the PSUs are modelled within the _module-types/Juniper_ folder.